yarn-project/foundation/src/serialize/serialize.ts

```diff
@@ -152,7 +152,7 @@ export function serializeToBufferArray(...objs: Bufferable[]): Buffer[] {
       ret.push(boolToBuffer(obj));
     } else if (typeof obj === 'bigint') {
       // Throw if bigint does not fit into 32 bytes
-      if (obj > BigInt('0xffffffffffffffffffffffffffffffff')) {
+      if (obj > BigInt('0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff')) {
```

Comment:
The value here was incorrect. `0xffffffffffffffffffffffffffffffff` is max value of 16 bytes, not 32.